From 437fd308c3a93c974a3363bedeb549e02d422929 Mon Sep 17 00:00:00 2001
From: Yi Yang <yangyi01@inspur.com>
Date: Fri, 7 Aug 2020 06:35:01 -0400
Subject: [PATCH 04/47] Update Documentation/topics/userspace-tso.rst

With GSO and GRO enabled, OVS DPDK can do GSO by software
if NIC can't support TSO or VXLAN TSO hardware offload.

Signed-off-by: Yi Yang <yangyi01@inspur.com>
---
 Documentation/topics/userspace-tso.rst | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/Documentation/topics/userspace-tso.rst b/Documentation/topics/userspace-tso.rst
index 14a7c6fb3..c7875fd99 100644
--- a/Documentation/topics/userspace-tso.rst
+++ b/Documentation/topics/userspace-tso.rst
@@ -87,8 +87,8 @@ used to enable same::
 Limitations
 ~~~~~~~~~~~
 
-The current OvS userspace `TSO` implementation supports flat and VLAN networks
-only (i.e. no support for `TSO` over tunneled connection [VxLAN, GRE, IPinIP,
+The current OvS userspace `TSO` implementation supports flat, VLAN and VXLAN
+networks only (i.e. no support for `TSO` over tunneled connection [GRE, IPinIP,
 etc.]).
 
 The NIC driver must support and advertise checksum offload for TCP and UDP.
@@ -98,11 +98,10 @@ in Open vSwitch. Currently, if the NIC supports that, then the feature is
 enabled, otherwise TSO can still be enabled but SCTP packets sent to the NIC
 will be dropped.
 
-There is no software implementation of TSO, so all ports attached to the
-datapath must support TSO or packets using that feature will be dropped
-on ports without TSO support.  That also means guests using vhost-user
-in client mode will receive TSO packet regardless of TSO being enabled
-or disabled within the guest.
+There is software implementation of TSO, which is called as GSO (Generic
+Segment Offload), so all ports attached to the datapath mustn't support TSO.
+That also means guests using vhost-user in client mode can receive TSO packet
+regardless of TSO being enabled or disabled within the guest.
 
 All kernel devices that use the raw socket interface (veth, for example)
 require the kernel commit 9d2f67e43b73 ("net/packet: fix packet drop as of
-- 
2.17.1

